### PR TITLE
Refactor to ensure we call check_consistency

### PR DESCRIPTION
**Pull request by erikjohnston**
The idea here is to stop people forgetting to call `check_consistency`. Folks can still just pass in `None` to the new args in `build_sequence_generator`, but hopefully they won't.

c.f.  #9382

### CODE REVIEW

**Comment by clokep** [comment]
Will probably want to merge the changes from #9449 into this?

**Comment by clokep** [comment]
I think this is related to #7938?

File: synapse/storage/databases/main/registration.py [hunk 1]
```diff
@@ -70,7 +70,12 @@ def _default_token_owner(self):
 
 
 class RegistrationWorkerStore(CacheInvalidationWorkerStore):
-    def __init__(self, database: DatabasePool, db_conn: Connection, hs: "HomeServer"):
+    def __init__(
+        self,
+        database: DatabasePool,
+        db_conn: LoggingDatabaseConnection,
```
Comment (clokep) [inline synapse/storage/databases/main/registration.py:76]: I remember you and @richvdh having a conversation about switching these from `Connection` to `LoggingDatabaseConnection`, but do not remember the OK. Are we OK making this change?

**Review by clokep — APPROVED** [review]
Looks like an improvement! Just one question (and I did wonder if `check_consistency` could be made private, but don't have a strong opinion on it).

File: synapse/storage/databases/main/registration.py (hunk 1 above)
Comment (erikjohnston) [reply synapse/storage/databases/main/registration.py:76]: Yeah, I think we are

Comment (richvdh) [reply synapse/storage/databases/main/registration.py:76]: he convinced me I think

**Pull request merged**